client/payment-methods/test/index.js

```diff
@@ -143,4 +144,40 @@ describe( 'PaymentMethods', () => {
 			'sofort',
 		] );
 	} );
+
+	test( 'express payments rendered when UPE preview feture flag is enabled', () => {
+		const featureFlagContext = {
+			featureFlags: { upeSettingsPreview: true, upe: false },
+		};
+
+		render(
+			<WCPaySettingsContext.Provider value={ featureFlagContext }>
+				<PaymentMethods />
+			</WCPaySettingsContext.Provider>
+		);
+
+		const enableWooCommercePaymentText = screen.getByText(
+			'Enable the new WooCommerce Payments checkout experience'
+		);
+
+		expect( enableWooCommercePaymentText ).toBeInTheDocument();
+	} );
+
+	test( 'express payments should not rendered when UPE preview feture flag is disabled', () => {
+		const featureFlagContext = {
+			featureFlags: { upeSettingsPreview: false, upe: false },
```

Comment:
How about also adding a test for the `upe = 1, upe preview = 1` ("UPE available but already enabled") scenario, and maybe also `upe = 1, upe preview = 0`?

You could refactor the second test that's already in place to use [`test.each()`](https://jestjs.io/docs/api#testeachtablename-fn-timeout) for this:

```suggestion
	test.each( [ [ false, false ], [ false, true ], [ true, true ] ] )( 'express payments should not rendered when UPE preview = %s and UPE = %s', ( upeSettingsPreview, upe ) => {
		const featureFlagContext = {
			featureFlags: { upeSettingsPreview, upe },
```

Reply:
I like the idea of using a data provider to test the combinations and their results! Thanks!